socs/agents/hwp_supervisor/agent.py

```diff
@@ -929,11 +1012,9 @@ def pid_to_freq(self, session, params):
             freq_tol_duration=params['freq_tol_duration'],
             direction=d
         )
-        success = self.control_state_machine.request_state(state)
-        if success:
-            return True, f"Set state to {state}"
-        else:
-            return False, "Failed to update state"
+        action = self.control_state_machine.request_new_action(state)
+        action.sleep_until_complete(session=session)
+        return True, f"Completed with state: {action.cur_state}"
```

Comment:
What happens when the requested action state fails? Does this return something like `True, f"Complete with state:" {'class': 'Error', 'msg': None, 'success': False}`?

Reply:
Yes... I can return `action.success` as the first param.

Reply:
Yeah, I think we should do that. Otherwise you need to inspect the `ControlState` that's returned to know whether the task completed successfully, rather than just `response.session['success']`.